core/modules/rproc/InfileMerger.cc

```diff
@@ -251,6 +267,56 @@ bool InfileMerger::_applySqlLocal(std::string const& sql) {
     return true;
 }
 
+
+size_t InfileMerger::_getResultTableSizeMB() {
+    std::string resultDb = _config.mySqlConfig.dbName;
+    std::string tableSizeSql = std::string("SELECT table_name, ")
+                             + "round(((data_length + index_length) / 1048576), 2) as 'MB' "
+                             + "FROM information_schema.TABLES "
+                             + "WHERE table_schema = '" + _config.mySqlConfig.dbName
+                             + "' AND table_name = '" + _mergeTable + "'";
+    LOGS(_log, LOG_LVL_DEBUG, "Checking ResultTableSize " << tableSizeSql);
+    std::lock_guard<std::mutex> m(_sqlMutex);
+    sql::SqlErrorObject errObj;
+    sql::SqlResults results;
+    if (not _sqlConn.get()) {
+        _sqlConn = std::make_shared<sql::SqlConnection>(_config.mySqlConfig, true);
+        if (not _sqlConn->connectToDb(errObj)) {
+            _error = util::Error(errObj.errNo(),
+                     _queryIdStr + "Error connecting to db: " + errObj.printErrMsg(),
+                     util::ErrorCode::MYSQLCONNECT);
+            _sqlConn.reset();
+            LOGS(_log, LOG_LVL_ERROR, "InfileMerger error: " << _error.getMsg());
+            return 0;
+        }
+        LOGS(_log, LOG_LVL_DEBUG, _queryIdStr <<"InfileMerger " << (void*) this << " connected to db");
+    }
```

Comment:
Looks like this whole `if `is copied from `InfileMerger::_applySqlLocal`, maybe move tit into separate method to avoid duplication?